### PR TITLE
e2e tests and management queries

### LINKED ISSUE #10: Support managements queries

### PR DESCRIPTION
**Pull request by roeap**
This PR got a bit larger then originally intended :). However I do hope it checks a few things of our list.

* re-add @AsafMah 's improvements to arrow code, which somehow got lost in merge to main
* Add a `ManagementQueryBuilder` and `execute_command` client method
* Introduce e2e tests using the `mock_transport_framework`
* Add CONTRIBUTING.md with explanation how to use  `mock_transport_framework`
* General housekeeping

closes #10

### CODE REVIEW

File: azure-kusto-data/src/operations/mgmt.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+use azure_core::prelude::*;
+use azure_core::setters;
+use azure_core::{collect_pinned_stream, Response as HttpResponse};
+use futures::future::BoxFuture;
+use serde::{Deserialize, Serialize};
+
+type ManagementQuery = BoxFuture<'static, crate::error::Result<KustoResponseDataSetV1>>;
+
+#[derive(Debug, Clone)]
+pub struct ManagementQueryBuilder {
```
Comment (AsafMah) [inline azure-kusto-data/src/operations/mgmt.rs:13]: Most of the code here is copy-pasted from query.rs.
I'm pretty sure we can do it in one generic class (I.E type ManagementQueryBuilder = QueryBuilder<ManagementQuery>)
Or at worst, having most of the functionality in one type, and have the QueryBuilders be thin wrappers around it

File: azure-kusto-data/src/client.rs [hunk 2]
```diff
@@ -112,8 +114,16 @@ impl KustoClient {
         ExecuteQueryBuilder::new(self.clone(), database.into(), query.into(), Context::new())
     }
 
-    pub(crate) fn prepare_request(&self, uri: &str, http_method: http::Method) -> Request {
-        let mut request = Request::new(uri.parse().unwrap(), http_method);
+    pub fn execute_command<DB, Q>(&self, database: DB, query: Q) -> ManagementQueryBuilder
```
Comment (AsafMah) [inline azure-kusto-data/src/client.rs:117]: Open question - in most SDKs we also expose a general "execute" method, that sends to either query on command based on the "." prefix.
Do we want it here?
It's pretty useful, but it's still technically a heuristic.

File: azure-kusto-data/tests/setup.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                TokenCredentialOptions::default(),
+            ));
+            (service_url, credential, database)
+        } else {
+            let credential = Arc::new(DummyCredential {});
+            let database = String::from_utf8_lossy(&std::fs::read(db_path).unwrap()).to_string();
+            (String::new(), credential, database)
+        };
+
+    let options = KustoClientOptions::new_with_transaction_name(transaction_name.to_string());
```
Comment (AsafMah) [inline azure-kusto-data/tests/setup.rs:66]: Can you explain a bit how the transactions work?
Do we actually send and receive data from the server, or is it all mocked with the request/response json files?
If it's all mocked, why do we need an actual client? Does it really count as e2e?

File: azure-kusto-data/src/operations/mgmt.rs (hunk 1 above)
Comment (roeap) [reply azure-kusto-data/src/operations/mgmt.rs:13]: Agreed, I tried doing it with asyc-convert, but there were some issues https://github.com/Azure/azure-sdk-for-rust/pull/685#discussion_r835772356 but there certainly is a way... wanted to leave this as a future optimization, but can try to tackle it now as well.

File: azure-kusto-data/src/client.rs (hunk 2 above)
Comment (roeap) [reply azure-kusto-data/src/client.rs:117]: Though about that as well. The one downside i see here is that we might have to wrap the responses in an enum for the V1 and V2 response sets. Generally i am leaning more towards offering that API though.

File: azure-kusto-data/tests/setup.rs (hunk 3 above)
Comment (roeap) [reply azure-kusto-data/tests/setup.rs:66]: The framework works in two modes - the RECORD and REPLAY mode. In Record mode there is just another pipeline element, that does the recording. This has to be executed locally against an actual service. Tests are then run in REPLAY mode, where no more communication with the server happens. However all steps are executed up until the transport level, where we do the communication with the service. Personally I would at least consider that e2e-ish :).

Comment (roeap) [reply azure-kusto-data/tests/setup.rs:66]: But ... does the Kusto team operate a cluster for use in CI/CD? if so that would be real end to end :)

Comment (AsafMah) [reply azure-kusto-data/tests/setup.rs:66]: We do :)
I'll implement it at some point

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "editor.formatOnSave": true
```
[the remark this answers is not in the document]
Comment (roeap) [reply .vscode/settings.json]: removed it.

**Review by AsafMah — APPROVED** [review]
Let's merge it :)

**Pull request merged**